week3/predict_citibike.Rmd

```diff
@@ -0,0 +1,391 @@
+---
+title: "predict_citibike"
+author: "Sushobhan Parajuli"
+date: "`r Sys.Date()`"
+output: html_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+```{r}
+library(tidyverse)
+library(dplyr)
+library(scales)
+library(modelr)
+library(lubridate)
+```
+
+```{r}
+# import the holidays date
+# join holidays with the trips_per_day
+
+holidays <- read_csv("US Bank holidays", col_names = c("row", "ymd", "holiday"))
+
+# import the data "trips_per_day.tsv"
+# mutate a column "is_weekend" and "is_holiday"
+trips_per_day <- read_tsv("trips_per_day.tsv") %>% 
+  left_join(holidays, by = "ymd") %>%
+  mutate(is_holiday = as.numeric(!is.na(holiday))) %>%
+  mutate(is_weekend = as.numeric(wday(ymd)==c(1,7))) %>%
+  select(-c(row, holiday))
+
+trips_per_day

```

Comment:
In the future try to have binary data recorded as Int or bool, sometimes the summary functions will act strangely if they see a variable is a double.